pkg/didcomm/protocol/introduce/states.go

```diff
@@ -53,11 +65,11 @@ func (s *noOp) CanTransitionTo(_ state) bool {
 	return false
 }
 
-func (s *noOp) ExecuteInbound(_ *metaData) (state, error) {
+func (s *noOp) ExecuteInbound(p provider, _ *metaData) (state, error) {
 	return nil, errors.New("cannot execute no-op")
 }
 
-func (s *noOp) ExecuteOutbound(_ *metaData, _ *service.Destination) (state, error) {
+func (s *noOp) ExecuteOutbound(p provider, _ *metaData, _ *service.Destination) (state, error) {
```

Comment:
Why are we passing provider in execute in bound and outbound ?

Reply:
the provider was equal to context (did-exchange), I have renamed it now (provider to internalContext)